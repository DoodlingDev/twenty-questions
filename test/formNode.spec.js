import React from "react";
import FormNode from "../src/components/formNode";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe("rendering", () => {
  it("renders string types correctly", () => {
    const render = renderer.create(
      <FormNode
        type="string"
        name="test_string"
        title="string title"
        description="A simple String"
        valueManager={{
          update: () => {},
          values: {},
          validate: {},
        }}
      />
    ).toJSON();
    expect(render).toMatchSnapshot();
  });
});
