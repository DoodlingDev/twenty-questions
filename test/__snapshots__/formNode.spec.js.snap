// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders string types correctly 1`] = `
<div
  className="q20_node__string"
>
  <span
    className="q20_nodeString__testString--description"
    htmlFor="undefined.test_string"
  >
    A simple String
  </span>
  <div
    className="q20_testString__errorHandler"
  >
    <input
      className="q20_nodeString__testString--input"
      id="undefined.test_string"
      onChange={[Function]}
      placeholder={undefined}
      type="text"
      value=""
    />
  </div>
</div>
`;
